tsconfig.json

```diff
@@ -18,7 +18,8 @@
     "lib": [
       "es5",
       "es2015",
-      "dom"
+      "dom",
+      "esnext" // test BaseVirtualPromise conforms with esnext

```

Comment:
I think it'll still work without requiring `esnext` lib explicitly in tsconfig? if so, I'd like exclude it.